tests/Unit/NumericalAlgorithms/LinearAlgebra/Test_FindGeneralizedEigenvalues.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "NumericalAlgorithms/LinearAlgebra/FindGeneralizedEigenvalues.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace {
+void test_find_generalized_eigenvalues(const Matrix& A, const Matrix& B,
+                                       const size_t eigenvector_size,
+                                       const size_t eigenvalue_size) {
+  // Set up vectors and a matrix to store the eigenvalues and eigenvectors
+  DataVector eigenvalues_real_part(eigenvalue_size, 0.0);
+  DataVector eigenvalues_im_part(eigenvalue_size, 0.0);
+  Matrix eigenvectors(eigenvector_size, eigenvector_size, 0.0);
+
+  // Get the generalized eigenvectors and eigenvalues
+  /// [generalized_eigenvalue_example]
+  find_generalized_eigenvalues(&eigenvalues_real_part, &eigenvalues_im_part,
+                               &eigenvectors, A, B);
+  /// [generalized_eigenvalue_example]
+
+  // Expected eigenvalues and eigenvectors
+  constexpr double expected_larger_eigenvalue = 10.099019513592784;
+  constexpr double expected_smaller_eigenvalue = -0.09901951359278449;
+  // We are testing a 2x2 matrix. Eigenvectors have two components.
+  // Normalize the eigenvectors so the second component is 1.
+  // Store the remaining components of the expected eigenvalues.
+  constexpr double expected_larger_eigenvector_value = 0.819803902718557;
+  constexpr double expected_smaller_eigenvector_value = -1.2198039027185574;
+
+  // Check eigenvalues
+  CHECK(eigenvalues_im_part[0] == approx(0.0));
+  CHECK(eigenvalues_im_part[1] == approx(0.0));
+  CHECK(max(eigenvalues_real_part) == approx(expected_larger_eigenvalue));
+  CHECK(min(eigenvalues_real_part) == approx(expected_smaller_eigenvalue));
+
+  // Check eigenvectors
+  if (eigenvalues_real_part[0] > eigenvalues_real_part[1]) {
+    CHECK(eigenvectors(0, 0) / eigenvectors(1, 0) ==
+          approx(expected_larger_eigenvector_value));
+    CHECK(eigenvectors(0, 1) / eigenvectors(1, 1) ==
+          approx(expected_smaller_eigenvector_value));
+  } else {
+    CHECK(eigenvectors(0, 0) / eigenvectors(1, 0) ==
+          approx(expected_smaller_eigenvector_value));
+    CHECK(eigenvectors(0, 1) / eigenvectors(1, 1) ==
+          approx(expected_larger_eigenvector_value));
+  }
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Numerical.LinearAlgebra.GeneralizedEigenvalue",
+                  "[NumericalAlgorithms][LinearAlgebra][Unit]") {
+  Matrix A(2, 2, 0.0);
```

Comment:
when you rebase (after #543 is merged) you can construct by passing the rows in...